packages/langium/src/generator/generator-node.ts

```diff
@@ -78,6 +89,14 @@ export class CompositeGeneratorNode {
         return this.contents.length === 0;
     }
 
+    trace<T extends AstNode>(astNode: T, property?: keyof T, index: number = -1): this {
```

Comment:
Suggestion: We have a special type for the `property` parameter here, that prevents users from accidentally using reserved property names such as `$type` or `$container`:
```suggestion
    trace<T extends AstNode>(astNode: T, property?: Properties<T>, index: number = -1): this {
```

Reply:
Good to know, thanks 👍